### ISSUE TITLE
Failed to create azure Cloud Credentials

### ISSUE THREAD

**Issue by kingsd041** [rancher/rancher#35600]
**Rancher Server Setup**
- Rancher version: v2.6.2
- Installation option (Docker install/Helm Chart):  Docker install
   - If Helm Chart, Kubernetes Cluster and version (RKE1, RKE2, k3s, EKS, etc):
- Proxy/Cert Details:

**Information about the Cluster**
- Kubernetes version: 
- Cluster Type (Local/Downstream):
   - If downstream, what type of cluster? (Custom/Imported or specify provider for Hosted/Infrastructure Provider):

**Describe the bug**

**To Reproduce**
 
1. Create Azure Cloud Credential
2. Select `AzureChinaCloud` for Environment, and other values are set correctly
3. Create

**Result**
Azure Cloud Credential failed to create, I can get the error log from the UI:

```
{"error":"could not find tenant ID: Unexpected response from Get Subscription: subscriptions.Client#Get: Failure responding to request: StatusCode=404 -- Original Error: autorest/azure: Service returned an error. Status=404 Code="SubscriptionNotFound" Message="The subscription 'xxxxxxxx-b2ae-4eb2-a69c-9ee96034073e' could not be found.""}
```
**Expected Result**

**Screenshots**

![image](https://user-images.githubusercontent.com/19684717/142591572-1749d1b6-924a-4701-a183-28611e63f4e0.png)

**Additional context**

Using the same `Tenant ID` `Client ID` `Client Secret` `Subscription ID` can be successfully created on v2.5.8

**Comment by christianrowlands** [comment]
I am running into the same error with Rancher v2.6.0.

I am trying to select the `AzureUSGovernmentCloud` option for the Cloud Credential Environment, and am experiencing exactly what @kingsd041 is seeing.

Worthy of note, I tried to add regular `AzurePublicCloud` credentials and it works as expected. However, when adding those regular Azure Public Cloud credentials I went ahead and tried selecting `AzureChinaCloud`, `AzureGermanCloud`, and `AzureUSGovernmentCloud` and all three worked as well. It is almost like the cloud selection drop down has no effect on testing the credentials.

To summarize, regular Azure Public Cloud credentials work just fine with any Environment selection, but Azure US Government Cloud credentials don't work with any of the Environment options selected.
